src/main/java/org/mule/extension/aggregator/internal/storage/content/SimpleAggregatedContent.java

```diff
@@ -53,15 +53,24 @@ public void add(TypedValue newContent, Long timeStamp) {
 
   @Override
   public void add(TypedValue newContent, Long timeStamp, int sequenceNumber) {
-    sequencedElements.put(sequenceNumber, newContent);
+    if (sequencedElements.containsKey(sequenceNumber)) {
+      sequencedElements.get(sequenceNumber).add(newContent);
+    } else {
+      List<TypedValue> newList = new ArrayList<>();
+      newList.add(newContent);
+      sequencedElements.put(sequenceNumber, newList);
+    }
+
+    //sequencedElements.put(sequenceNumber, newContent);
     updateTimes(timeStamp);
   }
 
   @Override
   public List<TypedValue> getAggregatedElements() {
     List<TypedValue> orderedElements = new ArrayList<>();
     if (sequencedElements.size() > 0) {
-      orderedElements = sequencedElements.entrySet().stream().sorted(comparingInt(Map.Entry::getKey)).map(Map.Entry::getValue)
+      orderedElements = sequencedElements.entrySet().stream().sorted(comparingInt(Map.Entry::getKey))
+          .flatMap(val -> val.getValue().stream())
```

Comment:
Is sorting necessary? if it, is not necessary to sort inside the lambda in the flatmap?

Reply:
The idea is to sort elements by sequence number. E.g. sequencedElements = { 4: [e, d, b], 2: [t, a, d], 3: [a, b, c] }
then { t, a, d, a, b, c, e, d, b } is returned.

Reply:
makes sense?

Reply:
yep